packages/plugins/response-cache/src/plugin.ts

```diff
@@ -558,13 +585,26 @@ export function useResponseCache<PluginContext extends Record<string, any> = {}>
 
       return handleMaybePromise(
         () =>
-          buildResponseCacheKey({
-            documentString: getDocumentString(onExecuteParams.args),
-            variableValues: onExecuteParams.args.variableValues,
-            operationName: onExecuteParams.args.operationName,
-            sessionId,
-            context: onExecuteParams.args.contextValue,
-          }),
+          buildResponseCacheKey(
+            new Proxy(
+              {
+                documentString: getDocumentString(onExecuteParams.args),
+                variableValues: onExecuteParams.args.variableValues,
+                operationName: onExecuteParams.args.operationName,
+                sessionId,
+                context: onExecuteParams.args.contextValue,
+                extras: undefined as any,
+              },
+              {
+                get: (obj, prop) => {
+                  if (prop === 'extras') {
+                    return getScopeFromQuery(schema, onExecuteParams.args.document.loc.source.body);
+                  }
+                  return obj[prop as keyof typeof obj];
+                },
+              },
+            ),
```

Comment:
I don't remembre how it was implemented before, but is there a reason to use a proxy instead of classic getters?

Like this: 
```suggestion
            {
              documentString: getDocumentString(onExecuteParams.args),
              variableValues: onExecuteParams.args.variableValues,
              operationName: onExecuteParams.args.operationName,
              sessionId,
              context: onExecuteParams.args.contextValue,
              get extras() {
                return getScopeFromQuery(schema, onExecuteParams.args.document.loc.source.body);
              },
            },
```

Reply:
A recent habit to use proxies for their flexibility, but I can see that a getter in this case would fit better for clarity sake.